Sources/TSCUtility/Version.swift

```diff
@@ -85,15 +92,17 @@ public enum VersionError: Error, CustomStringConvertible {
 }
 
 extension Version {
-    // TODO: Rename this function to `init(string: String) throws`, after `init?(string: String)` is removed.
+    // TODO: Rename this function to `init(string:usesLenientParsing:) throws`, after `init?(string: String)` is removed.
     // TODO: Find a better error-checking order.
     // Currently, if a version string is "forty-two", this initializer throws an error that says "forty" is only 1 version core identifier, which is not enough.
     // But this is misleading the user to consider "forty" as a valid version core identifier.
     // We should find a way to check for (or throw) "wrong characters used" errors first, but without overly-complicating the logic.
     /// Creates a version from the given string.
-    /// - Parameter versionString: The string to create the version from.
+    /// - Parameters:
+    ///   - versionString: The string to create the version from.
+    ///   - usesLenientParsing: A Boolean value indicating whether or not the version string should be parsed leniently. If `true`, then the patch version is assumed to be `0` if it's not provided in the version string; otherwise, the parsing strictly follows the Semantic Versioning 2.0.0 rules. This value defaults to `false`.
     /// - Throws: A `VersionError` instance if the `versionString` doesn't follow [SemVer 2.0.0](https://semver.org).
-    public init(versionString: String) throws {
+    public init(versionString: String, usesLenientParsing: Bool = false) throws {
```

Comment:
iirc we needed two constructors instead of one with default value because of downstream dependencies on the exact signatures (passed into map{})

Reply:
I think the one that downstream dependencies use is `init?(string:)`. This initializer still exists and calls `init?(_:)`, which then calls `init(versionString: versionString, usesLenientParsing: false) throws`.

Reply:
I have the change ready to commit, but also had the thought the downstream dependencies probably use the deprecated function. I could see the use in having a new version ready to cut over to, but that might get into project planning that I don't know the scope of.

Reply:
thanks @colincornaby and @WowbaggersLiquidLunch - lets get this merged first, then @colincornaby you can submit a PR to SwiftPM to use the new methods where appropriate.

Reply:
Sounds good! Thanks!